runners_manager/runner/Runner.py

```diff
@@ -80,21 +89,40 @@ def toJson(self):
 
         return d
 
-    def update_status(self, elem):
-        if elem['status'] == 'online' and elem['busy'] is True:
-            status = 'running'
-        else:
-            status = elem['status']
-
+    def update_status(self, status: str):
         if self.status == status:
             return
 
-        if self.status == 'offline' and status != 'offline':
+        if self.status not in ['online', 'running'] and status in ['online', 'running']:
             self.started_at = datetime.datetime.now()
 
         self.status_history.append(self.status)
+
+        logger.info(f'Runner {self.name} updating status from {self.status} to {status}')
         self.status = status
 
+        PROMETHEUS_RUNNER_STATUS.labels(
+            name=self.name,
+            flavor=self.vm_type.flavor,
+            image=self.vm_type.image
+        ).state(self.status)
+
+        # TODO: Find why the __delete__ function is not called or why __del__ is called all the time.
+        if self.status == 'deleting':
+            PROMETHEUS_RUNNER_STATUS.remove(self.name, self.vm_type.flavor, self.vm_type.image)
```

Comment:
I need help figuring it out this one @erwan-b, if you have some free time in your day. Ideally I want to use the `__del__` or `__delete__` class to perform this call

Reply:
I would prefer not.

As we are using a database to store the runners, the object is the a representation of the database state, the object is not the runner.
There for, the object runner can potentially be created / deleted regardless the database state.

Reply:
Understood. I'll keep the following logic then.